dbms/src/DataStreams/FilterBlockInputStream.cpp

```diff
@@ -170,9 +174,18 @@ Block FilterBlockInputStream::readImpl()
                 current_column.column = current_column.column->filter(*filter_and_holder.data, -1);
         }
 
-        return res;
+        return removeFilterIfNeed(std::move(res));
     }
 }
 
 
+Block FilterBlockInputStream::removeFilterIfNeed(Block && block)
+{
+    if (block && remove_filter)
+        block.erase(static_cast<size_t>(filter_column));
+
+    return block;
```

Comment:
Better to add std::move here otherwise a copy will be performed (the compiler won't do copy elision for the parameter).